smart-frontend/app/src/store/adh/rules/rulesActionsSlice.ts

```diff
@@ -0,0 +1,177 @@
+ * limitations under the License.
+ */
+import { createCrudSlice } from '@store/redux/createCrudSlice';
+import type { AdhRule } from '@models/adh';
+import type { ModalState } from '@models/modal';
+import { createAsyncThunk } from '@store/redux';
+// eslint-disable-next-line import/no-cycle
+import { getRules } from './rulesSlice';
+import { showError, showSuccess } from '@store/notificationsSlice';
+import { getErrorMessage } from '@utils/responseUtils';
+import type { RequestError } from '@api';
+import { AdhRulesApi } from '@api';
+import type { PayloadAction } from '@reduxjs/toolkit';
+
+const createRule = createAsyncThunk('adh/rulesActions/createRule', async (text: string, thunkAPI) => {
+  thunkAPI.dispatch(setIsActionInProgress(true));
+  thunkAPI.dispatch(closeCreateRuleDialog());
+  try {
+    await AdhRulesApi.createRule(text);
+    thunkAPI.dispatch(showSuccess({ message: 'New rule was created and applied successfully' }));
+  } catch (error) {
+    thunkAPI.dispatch(showError({ message: getErrorMessage(error as RequestError) }));
+    return thunkAPI.rejectWithValue(error);
+  } finally {
+    thunkAPI.dispatch(setIsActionInProgress(false));
+  }
+});
+
+const deleteRule = createAsyncThunk('adh/rulesActions/deleteRule', async (ruleId: number, thunkAPI) => {
+  thunkAPI.dispatch(setIsActionInProgress(true));
+  thunkAPI.dispatch(closeDeleteRuleDialog());
+  try {
+    await AdhRulesApi.deleteRule(ruleId);
+    thunkAPI.dispatch(showSuccess({ message: `Rule #${ruleId} was deleted` }));
+  } catch (error) {
+    thunkAPI.dispatch(showError({ message: getErrorMessage(error as RequestError) }));
+    return thunkAPI.rejectWithValue(error);
+  } finally {
+    thunkAPI.dispatch(setIsActionInProgress(false));
+  }
+});
+
+const startRule = createAsyncThunk('adh/rulesActions/startRule', async (ruleId: number, thunkAPI) => {
+  thunkAPI.dispatch(closeStartRuleDialog());
+  try {
+    await AdhRulesApi.startRule(ruleId);
+    thunkAPI.dispatch(showSuccess({ message: `Rule #${ruleId} was started successfully` }));
+  } catch (error) {
+    thunkAPI.dispatch(showError({ message: getErrorMessage(error as RequestError) }));
+    return thunkAPI.rejectWithValue(error);
+  }
+});
+
+const stopRule = createAsyncThunk('adh/rulesActions/stopRule', async (ruleId: number, thunkAPI) => {
+  thunkAPI.dispatch(closeStopRuleDialog());
+  try {
+    await AdhRulesApi.stopRule(ruleId);
+    thunkAPI.dispatch(showSuccess({ message: `Rule #${ruleId} was stopped successfully` }));
+  } catch (error) {
+    thunkAPI.dispatch(showError({ message: getErrorMessage(error as RequestError) }));
+    return thunkAPI.rejectWithValue(error);
+  }
+});
+
+const withUpdate = (name: string, dialogAction: ReturnType<typeof createAsyncThunk>) => {
+  return createAsyncThunk(name, async (payload: unknown, thunkAPI) => {
+    await thunkAPI.dispatch(dialogAction(payload)).unwrap();
+    await thunkAPI.dispatch(getRules());
+  });
+};
```

Comment:
can we make this very function a global one? 

we face cases when we need to update pages once we create/update/delete stuff all the time. instead of copy/paste the func, we can import it. we can also make use of it in other projects

for example, we can store it in the same folder where we store ```createCrudSlice```